mail_tracking/models/ir_mail_server.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+# © 2016 Antonio Espinosa - <antonio.espinosa@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import re
+from openerp import models, api, tools
+
+
+class IrMailServer(models.Model):
+    _inherit = "ir.mail_server"
+
+    def _tracking_headers_add(self, tracking_email_id, headers):
+        """Allow other addons to add its own tracking SMTP headers"""
+        headers = headers or {}
+        headers['X-Odoo-Tracking-ID'] = tracking_email_id
+        return headers
+
+    def _tracking_email_id_body_get(self, body):
+        body = body or ''
+        tracking_email_id = False
+        # https://regex101.com/r/lW4cB1/2
+        match = re.search(
+            r'<img [^>]* data-odoo-tracking-email=["\']([0-9]*)["\']', body)
+        if match:
+            try:
+                tracking_email_id = int(match.group(1))
+            except:
+                pass
+        return tracking_email_id
+
+    def build_email(self, email_from, email_to, subject, body, email_cc=None,
+                    email_bcc=None, reply_to=False, attachments=None,
+                    message_id=None, references=None, object_id=False,
+                    subtype='plain', headers=None, body_alternative=None,
+                    subtype_alternative='plain'):
+        tracking_email_id = self._tracking_email_id_body_get(body)
+        if tracking_email_id:
+            headers = self._tracking_headers_add(tracking_email_id, headers)
+        msg = super(IrMailServer, self).build_email(
+            email_from, email_to, subject, body, email_cc=email_cc,
+            email_bcc=email_bcc, reply_to=reply_to, attachments=attachments,
+            message_id=message_id, references=references, object_id=object_id,
+            subtype=subtype, headers=headers,
+            body_alternative=body_alternative,
+            subtype_alternative=subtype_alternative)
+        return msg
+
+    def _tracking_email_get(self, message):
+        tracking_email_id = False
+        try:
```

Comment:
Instead of try, you can use `message['X-Odoo-Tracking-ID'].isdigit()` to see if it's a number.